test/built-ins/Atomics/wait/cannot-suspend-throws.js

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2018 Amal Hussein. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
```

Comment:
@nomadtechie covers this part: 

> when [[CanBlock]] is false for an agent: call Atomics.wait on the main thread and expect that it will throw a TypeError since you can not block on the main thread